audio/src/lib.rs

```diff
@@ -116,18 +116,10 @@
 //! [audio-device]: https://docs.rs/audio-device
 //! [audio-generator]: https://docs.rs/audio-generator
 //! [audio]: https://docs.rs/audio
-//! [Buf]: https://docs.rs/audio-core/latest/audio_core/trait.Buf.html
-//! [BufMut]: https://docs.rs/audio-core/latest/audio_core/trait.BufMut.html
 //! [cpal]: https://github.com/RustAudio/cpal
-//! [Dynamic::resize]: https://docs.rs/audio/latest/audio/dynamic/struct.Dynamic.html#method.resize
-//! [dynamic!]: https://docs.rs/audio/latest/audio/macros/macro.dynamic.html
-//! [Dynamic]: https://docs.rs/audio/latest/audio/dynamic/struct.Dynamic.html
-//! [Interleaved]: https://docs.rs/audio/latest/audio/interleaved/struct.Interleaved.html
 //! [minimp3-rs]: https://github.com/germangb/minimp3-rs
 //! [play-mp3]: https://github.com/udoprog/audio/tree/main/examples/src/bin/play-mp3.rs
 //! [rubato]: https://github.com/HEnquist/rubato
-//! [Sequential]: https://docs.rs/audio/latest/audio/sequential/struct.Sequential.html
-//! [wrap]: https://docs.rs/audio/latest/audio/wrap/index.html
```

Comment:
Need to keep this for README (it is generated from this source) links to work.

Reply:
Oof, that's problematic. The reason I got rid of these is that it got very confusing when working on the documentation locally because I unexpectedly got linked from my local build of the documentation to the outdated docs on docs.rs.

Reply:
What tool do you use to generate README.md?

Reply:
`cargo-readme` initially, but I've switched to a custom project tool I'm using (which also validates more things).

Reply:
Is that published somewhere? Maybe we could make it automatically convert intradoc links to links to docs.rs in the README?

Reply:
I've tried to do it, but that would require accessing type information like rustdoc does. It's available (as json) but very mangled and hard to process. For now, this is the easiest way.

Reply:
I realized another problem with this approach of hardcoding links to the latest docs is that old versions of the docs will be broken because they will surprisingly jump to the latest version.

Reply:
What I've done for other Rust libraries is have a manually written README.md that is very brief and mostly exists to link to docs.rs, for example: https://github.com/PolyMeilex/rfd/ That approach doesn't require regular updates to the README.

Reply:
Having a rich README with basic instructions saves the click through and managing the links isn't enough of a hassle to sacrifice it. Just keep the links as-is and add any updated ones. Eventually I might get around to giving my tool support for generating type links for it, at which point I'll release it (since it does more stuff than `cargo-readme`) and these can be systematically removed.